app/src/main/java/me/rocka/fcitx5test/ui/common/DialogSeekBarPreference.kt

```diff
@@ -0,0 +1,119 @@
+package me.rocka.fcitx5test.ui.common
+
+import android.content.Context
+import android.util.AttributeSet
+import android.view.LayoutInflater
+import android.widget.SeekBar
+import androidx.appcompat.app.AlertDialog
+import androidx.preference.Preference
+import androidx.preference.PreferenceManager
+import me.rocka.fcitx5test.R
+import me.rocka.fcitx5test.databinding.DialogSeekBarBinding
+
+/**
+ * Custom preference which represents a seek bar which shows the current value in the summary. The
+ * value can be changed by clicking on the preference, which brings up a dialog which a seek bar.
+ * This implementation also allows for a min / max step value, while being backwards compatible.
+ *
+ * @property defaultValue The default value of this preference.
+ * @property min The minimum value of the seek bar. Must not be greater or equal than [max].
+ * @property max The maximum value of the seek bar. Must not be lesser or equal than [min].
+ * @property step The step in which the seek bar increases per move. If the provided value is less
+ *  than 1, 1 will be used as step.
+ * @property unit The unit to show after the value. Set to an empty string to disable this feature.
+ */
+class DialogSeekBarPreference : Preference {
+    var defaultValue: Int = 0
+    var min: Int = 0
+    var max: Int = 100
+    private var step: Int = 1
+    private var unit: String = ""
+
+    @Suppress("unused")
+    constructor(context: Context) : this(context, null)
+    constructor(context: Context, attrs: AttributeSet?) :
+            this(context, attrs, R.attr.preferenceStyle)
+    constructor(context: Context, attrs: AttributeSet?, defStyleAttr: Int) :
+            super(context, attrs, defStyleAttr) {
+        onPreferenceChangeListener = OnPreferenceChangeListener { _, newValue ->
+            summary = getTextForValue(newValue.toString())
+            true
+        }
+        onPreferenceClickListener = OnPreferenceClickListener {
+            showSeekBarDialog()
+            true
+        }
+    }
+
+    override fun onAttachedToHierarchy(preferenceManager: PreferenceManager?) {
+        super.onAttachedToHierarchy(preferenceManager)
+        summary = getTextForValue(
+            preferenceDataStore?.getInt(key, defaultValue) ?: defaultValue
```

Comment:
It would be better to introduce a function for this getting. The same thing repeats at L71 and L92.

Reply:
I see, I'll fix it soon.